fpmsyncd/routesync.cpp

```diff
@@ -281,23 +279,67 @@ void RouteSync::onVnetRouteMsg(int nlmsg_type, struct nl_object *obj, string vne
  */
 bool RouteSync::getIfName(int if_index, char *if_name, size_t name_len)
 {
+    struct nl_sock *sk = NULL;
+    struct nl_cache *cache = NULL;
+    int err;
+
     if (!if_name || name_len == 0)
     {
         return false;
     }
 
     memset(if_name, 0, name_len);
+    
+    sk = nl_socket_alloc();
+    if (!sk) 
+    {
+        SWSS_LOG_ERROR("Unable to allocate netlink socket");
+        return false;
+    }
+
+    err = nl_connect(sk, NETLINK_ROUTE);
+    if (err < 0)
+    {
+        SWSS_LOG_ERROR("Unable to connect netlink socket: %s", nl_geterror(err));
+        nl_socket_free(sk);
+        return false;
+    }
+
+    err = rtnl_link_alloc_cache(sk, AF_UNSPEC, &cache);
+    if (err < 0)
+    {
+        SWSS_LOG_ERROR("Unable to allocate link cache: %s", nl_geterror(err));
+        nl_close(sk);
+        nl_socket_free(sk);
+        return false;
+    }
 
     /* Cannot get interface name. Possibly the interface gets re-created. */
-    if (!rtnl_link_i2name(m_link_cache, if_index, if_name, name_len))
+    if (!rtnl_link_i2name(cache, if_index, if_name, name_len))
     {
-        rtnl_link_alloc_cache(m_nl_sock, AF_UNSPEC, &m_link_cache);
```

Comment:
what about using` nl_cache_refill` method instead of  alloc and and free repeatedly?